src/linear_algebra/finsupp.lean

```diff
@@ -371,10 +371,39 @@ begin
     refine sum_mem _ _, simp [this] }
 end
 
-theorem mem_span_iff_total {s : set α} {x : M}:
+theorem mem_span_iff_total {s : set α} {x : M} :
   x ∈ span R (v '' s) ↔ ∃ l ∈ supported R R s, finsupp.total α M R v l = x :=
 by rw span_eq_map_total; simp
 
+lemma exists_finset_of_mem_supr {ι : Type*} (p : ι → submodule R M) {m : M} (hm : m ∈ ⨆ i, p i) :
+  ∃ s : finset ι, m ∈ ⨆ i ∈ s, p i :=
+begin
+  obtain ⟨f, hf, rfl⟩ : ∃ f ∈ supported R R (⋃ i, ↑(p i)), finsupp.total M M R id f = m,
+  { have aux : (id : M → M) '' (⋃ (i : ι), ↑(p i)) = (⋃ (i : ι), ↑(p i)) := set.image_id _,
+    rwa [supr_eq_span, ← aux, finsupp.mem_span_iff_total R] at hm },
+  let t : finset M := f.support,
+  have ht : ∀ x : {x // x ∈ t}, ∃ i, x.1 ∈ p i,
+  { intros x,
+    rw finsupp.mem_supported at hf,
+    specialize hf x.2,
+    rwa set.mem_Union at hf },
+  rcases classical.axiom_of_choice ht with ⟨g, hg⟩,
+  let s : finset ι := finset.image g finset.univ,
+  use s,
+  show _ ∈ lattice.Inf _,
```

Comment:
It would really help readability if you would specify the contents of the underscores here. Also, I guess the next proof block should be between braces (and also the one after the second `show`).

Reply:
This show is only there to do some definitional unfolding. The stuff on the second `_` is rather ugly. Maybe part of this proof should be factored into a cleaner sublemma. I'll take a look.
The second `show` wasn't necessary at all. It replaced an `id x` by `x`.

Reply:
I'm working on getting the right sublemma. So I'll mark this PR as WIP for the time being.

Reply:
@sgouezel I've cleaned up the proofs some more.